tests/test_build.py

```diff
@@ -1,248 +1,154 @@
 from pathlib import Path
-from subprocess import run, PIPE
+
 import pytest
-from bs4 import BeautifulSoup as bs
+from bs4 import BeautifulSoup
+
+from jupyter_book import commands
 
 
-path_tests = Path(__file__).parent.resolve()
-path_books = path_tests.joinpath("books")
-path_root = path_tests.parent
+def test_create(tmpdir, cli):
+    book = Path(tmpdir) / "new_book"
+    result = cli.invoke(commands.create, str(book))
+    assert result.exit_code == 0
+    assert book.joinpath("_config.yml").exists()
+    assert len(list(book.iterdir())) == 9
 
 
-def test_build_book(tmpdir):
+def test_build_from_template(tmpdir, cli):
     """Test building the book template and a few test configs."""
     # Create the book from the template
-    path = Path(tmpdir).joinpath("mybook").absolute()
-    run(f"jb create {path}".split())
-
-    # Ensure the book is created properly
-    assert path.joinpath("_config.yml").exists()
-
-    # Build the book
-    run(f"jb build {path}".split(), check=True)
-    path_html = path.joinpath("_build", "html")
-    assert path_html.joinpath("index.html").exists()
-    assert path_html.joinpath("intro.html").exists()
-
-    # Test custom config values
-    path_config = path_books.joinpath("config")
-    run(f"jb build {path_config}".split(), check=True)
-    html = path_config.joinpath("_build", "html", "index.html").read_text(
-        encoding="utf8"
-    )
+    book = Path(tmpdir) / "new_book"
+    _ = cli.invoke(commands.create, str(book))
+    build_result = cli.invoke(commands.build, str(book))
+    assert build_result.exit_code == 0
+    html = book.joinpath("_build", "html")
+    assert html.joinpath("index.html").exists()
+    assert html.joinpath("intro.html").exists()
+
+
+def test_custom_config(cli, build_resources):
+    books, _ = build_resources
+    config = books.joinpath("config")
+    result = cli.invoke(commands.build, str(config))
+    assert result.exit_code == 0
+    html = config.joinpath("_build", "html", "index.html").read_text()
     assert '<h1 class="site-logo" id="site-title">TEST PROJECT NAME</h1>' in html
     assert '<div class="sphinx-tabs docutils container">' in html
     assert '<link rel="stylesheet" type="text/css" href="_static/mycss.css" />' in html
     assert '<script src="_static/js/myjs.js"></script>' in html
 
 
-def test_toc_builds(tmpdir):
+@pytest.mark.parametrize("toc", ["_toc.yml", "_toc_startwithlist.yml"])
```

Comment:
@choldgraf , there is one expected test failure caused by this ToC file.  Is disabling warnings here OK? The warning is `Found a content page that is not in _toc.yml: content3.rst.`

Reply:
yep that is fine with me if it is expected!

Reply:
To clarify, I'm saying that the failure was expected in the sense that I personally knew that this would fail because it failed locally.  And it's failing because of a warning.  We can get it to not fail by removing the `-W` flag.  So I just wanted to check with you to make sure you were good with removing the `-W` flag in this particular unit test.

Reply:
why is content3.rst not present? Should it be added to build_resources or removed from _toc.yml?

Reply:
IIRC this is probably an inefficiency in our testing infrastructure, where we use a single book template for tests and in some cases only want to include a page in the build of one of the tests but not others. Could be wrong there though

Reply:
if we don't want to fix it here, I would at least but a TODO comment that this should be fixed at a later date

Reply:
yep totally agree 👍

Reply:
`content3.rst` is present on disk, just not in the yaml file. I think `content3.rst` actually needs to be added to `_toc.yml` for this particular test.  

❗  I just noticed that `tests_toc.py` is actually modifying `_toc.yml` _on disk_, in my cwd repo (not in a temp dir).  In other words, `tests_toc.py` is mutating test data, the shape of which other tests depend upon.  This is where the new fixtures in `conftest.py` will come in handy.  Interestingly, `tests_toc.py` doesn't even care if `content3.rst` is in `_toc.yml`.  

Perhaps `./tests/books/toc/` should be restructured such that sub dirs represent books tailor made for specific test cases. The entire root can be copied to a temp dir/returned by the build_resources fixture for each test, and tests can specify which book they want.

Reply:
That sounds a bit like what I do in https://github.com/executablebooks/MyST-Parser/tree/master/tests/test_sphinx. Haven't looked at what @choldgraf has done with the testing infrastructure here yet though. I may have notes 😉

Reply:
BTW, this was leading to indeterminate tests, e.g., `test_toc_builds` would pass on the first run and fail on the second (but only when `test_toc.py` was run).  😭 

I'll introduce the minimal changeset to avoid this problem so that we can just get this merged and we can take a deeper look later.  Please see 662cefd.

Reply:
cheers, I created an issue for it #857

Reply:
😬 I can promise you that the jupyter book test infra is not the optimized lamborghini of testing that @chrisjsewell generally constructs. Sorryyyyy!

Reply:
it's almost like you guys are far better programmers than I am 🤔🤔🤔